performance_test/src/experiment_configuration/experiment_configuration.hpp

```diff
@@ -150,18 +176,35 @@ class ExperimentConfiguration
     m_roundtrip_mode(RoundTripMode::NONE)
   {}
 
+#ifdef ODB_FOR_SQL_ENABLED
+  friend class odb::access;
+#endif
+
   /// Throws #std::runtime_error if the experiment is not set up.
   void check_setup() const;
 
   /// Generates filename from the experiment configuration and opens a file accordingly. This will
   /// throw if the experiment configuration is not set up.
   void open_file();
 
+  // Using the GUID of the experiment as ID.
+#ifdef ODB_FOR_SQL_ENABLED
+  #pragma db id
+#endif
   boost::uuids::uuid m_id;
+#ifdef ODB_FOR_SQL_ENABLED
+  #pragma db transient
+#endif
   bool m_is_setup;
-
+#ifdef ODB_FOR_SQL_ENABLED
+  #pragma db transient
+#endif
   std::string m_logfile;
   std::string m_final_logfile_name;
+
+#ifdef ODB_FOR_SQL_ENABLED
+  #pragma db transient
+#endif
```

Comment:
Do we need to include these multiple times?

Reply:
I don't include anything here: #pragma db transient has to be over the variable that odb compiler should ignore (so it won't go to database), like bool m_is_setup,  std::string m_logfile, mutable std::ofstream m_os in our case.

Reply:
Okay maybe you know more about this better but instead of :
```
#ifdef ODB_FOR_SQL_ENABLED
  #pragma db transient
#endif
  bool m_is_setup;

#ifdef ODB_FOR_SQL_ENABLED
  #pragma db transient
#endif
  std::string m_logfile;
  std::string m_final_logfile_name;

#ifdef ODB_FOR_SQL_ENABLED
  #pragma db transient
#endif
```
Can we do this :
```
#ifdef ODB_FOR_SQL_ENABLED
  #pragma db transient
#endif
  bool m_is_setup;
  std::string m_logfile;
  std::string m_final_logfile_name;

```
Correct me if I am wrong.

Reply:
No, you can't do this. See for example `m_final_logfile_name` - it's stored in the database although above `m_logfile` is made `transient` and it's not saved. So `#pragma db transient` works only for the variable below.

Reply:
Okay thank you for the clarification!